Engine/Results/BaseResultsHandler.cs

```diff
@@ -509,6 +525,89 @@ protected virtual void SampleBenchmark(DateTime time, decimal value)
             Sample("Benchmark", "Benchmark", 0, SeriesType.Line, time, value);
         }
 
+        /// <summary>
+        /// Sample drawdown of equity of the strategy
+        /// </summary>
+        /// <param name="time">Time of the sample</param>
+        /// <param name="currentPortfolioValue">Current equity value</param>
+        protected virtual void SampleDrawdown(DateTime time, decimal currentPortfolioValue)
+        {
+            // This will throw otherwise, in this case just don't sample
+            if (CumulativeMaxPortfolioValue != 0)
+            {
+                // Calculate our drawdown and sample it
+                var drawdown = Statistics.Statistics.DrawdownPercent(currentPortfolioValue, CumulativeMaxPortfolioValue);
+                Sample("Drawdown", "Equity Drawdown", 0, SeriesType.Line, time, drawdown, "%");
+            }
+        }
+
+        /// <summary>
+        /// Sample assets sales volume
+        /// </summary>
+        /// <param name="time">Time of the sample</param>
+        protected virtual void SampleSalesVolume(DateTime time)
+        {
+            foreach (var holding in Algorithm.Portfolio.Values)
+            {
+                if (holding.TotalSaleVolume != 0)
```

Comment:
`if (holding.TotalSaleVolume != 0)` + We might want to order by TotalSaleVolume and cap the quantity being sampled, thinking about an algorithm trading multiple equities options, securities will grow quite big in time, say QC500 + options